client/src/components/JurisdictionAdmin/BatchRoundDataEntry/BatchDetails.tsx

```diff
@@ -0,0 +1,472 @@
+      />
+    </Detail>
+  )
+}
+
+interface IBatchResultTallySheetProps {
+  addSheet: () => Promise<void>
+  areResultsFinalized: boolean
+  batch: IBatch
+  contest: IContest
+  deleteSheet: () => Promise<void>
+  isEditing: boolean
+  selectedTabId: string
+  setAreChangesUnsaved: (areChangesUnsaved: boolean) => void
+  setIsEditing: (isEditing: boolean) => void
+  sheets: IBatchResultTallySheetWithId[]
+  updateSheet: (updatedSheet: IBatchResultTallySheet) => Promise<void>
+
+  // Require a key to ensure that the state within this component resets when a different sheet is
+  // selected
+  key: string // eslint-disable-line react/no-unused-prop-types
+}
+
+const BatchResultTallySheet: React.FC<IBatchResultTallySheetProps> = ({
+  addSheet,
+  areResultsFinalized,
+  batch,
+  contest,
+  deleteSheet,
+  isEditing,
+  selectedTabId,
+  setAreChangesUnsaved,
+  setIsEditing,
+  sheets,
+  updateSheet,
+}) => {
+  const tabs = tabsFromSheets(sheets)
+  const isTotalsSheet = selectedTabId === VOTE_TOTALS_TAB_ID
+  const selectedSheet = isTotalsSheet
+    ? undefined
+    : sheets.find(sheet => sheet.id === selectedTabId)
+
+  const formMethods = useForm<IBatchResultTallySheet>({
+    defaultValues: selectedSheet,
+  })
+  const { errors, formState, handleSubmit, register, reset } = formMethods
+  // Important gotcha! You have to access properties on the formState to subscribe to it:
+  // https://github.com/react-hook-form/react-hook-form/issues/9002
+  const { isSubmitting, isDirty } = formState
+
+  // Communicate up to the parent whether or not there are unsaved changes
+  useEffect(() => {
+    setAreChangesUnsaved(isDirty)
+  }, [isDirty, setAreChangesUnsaved])
+
+  const discardChanges = () => {
+    reset(selectedSheet)
+    setIsEditing(false)
+  }
+
+  const onValidSubmit: SubmitHandler<IBatchResultTallySheet> = async sheet => {
+    try {
+      await updateSheet(sheet)
+    } catch {
+      // Errors are automatically toasted by the queryClient
+      return
+    }
+    // Reset the form's isDirty value back to false
+    reset(sheet)
+    setIsEditing(false)
+  }
+
+  return (
+    <>
+      {isEditing && sheets.length > 1 ? (
+        // A special tab bar with a sheet name form input replacing the selected tab, rendered here
+        // instead of in the parent component so that we can create a separate react-hook-form form
+        // per sheet (sharing the form across sheets requires diligent resetting)
+        <Tabs id={batch.name} selectedTabId={selectedTabId}>
+          {tabs.map(tab =>
+            selectedTabId === tab.id ? (
+              <input
+                aria-label="Sheet Name"
+                className={classnames(
+                  Classes.INPUT,
+                  errors.name && Classes.INTENT_DANGER
+                )}
+                key={tab.id}
+                name="name"
+                readOnly={isSubmitting}
+                ref={register({
+                  required: true,
+                })}
```

Comment:
I'm no longer checking that the sheet name is unique client-side because there just isn't enough space for "Must be unique" error text. And making the input outline red without any explanatory text (as we do for this and other inputs when empty) feels like it wouldn't provide enough information to the user.

<img width="665" alt="error" src="https://user-images.githubusercontent.com/12616928/199046523-f15d3f53-7366-431f-9720-16d7637b2322.png">

Reply:
Great thinking, I think this makes a lot of sense. We can tweak the server-side error message to be a bit more friendly to this case.

Reply:
Tweaked! https://github.com/votingworks/arlo/pull/1661/commits/84d792100272ef25c10241e95dc1ac4049b1ed04